### ISSUE TITLE
switching from code to block or block to code will destroy nested if-else

### ISSUE THREAD

**Issue by Kirua-Sama** [Entkenntnis/robot-karol-online#707]
When creating nested if - else
and switching mode, either from block to code or from code to block
the "else" will be removed. 
and is_brick(1) will be replaced to is_brick(2)

please refer to picture. 
![before](https://github.com/Entkenntnis/robot-karol-online/assets/166800471/5191c2e0-db04-4871-9639-57e5930e802d)
![after](https://github.com/Entkenntnis/robot-karol-online/assets/166800471/f2821a5b-a134-46b0-a078-35199ccdf63a)

**Comment by Entkenntnis** [comment]
Will look into it, thanks for the report

**Comment by Entkenntnis** [comment]
I can reproduce with your example

```
command beforeUnmark
end_command

command oneOrTwoBrick
  if is_brick(1) then
    turn_right
    beforeUnmark
  else
    if is_brick(2) then
      turn_right(2)
      beforeUnmark
    else
      step
    end_if
  end_if
end_command
```

**Comment by Entkenntnis** [comment]
the is_brick(n) issue is solved, still investigating the else issue

**Comment by Entkenntnis** [comment]
minimal example:

```
if is_brick(10) then
  step
else
  step
end_if
```

**Comment by Entkenntnis** [comment]
This bug existsed for 2 years and is finally fixed in 7e87e4ba76c9a9aa20900292adc4c8ec0df6ada3
